src/sqlfluff/dialects/dialect_databricks.py

```diff
@@ -348,7 +348,63 @@
 )
 
 
-# Object References
+class IdentifierClauseSegment(BaseSegment):
+    """An `IDENTIFIER` clause segment.
+
+    https://docs.databricks.com/en/sql/language-manual/sql-ref-names-identifier-clause.html
+    """
+
+    type = "identifier_clause_segment"
+    match_grammar = Sequence(
+        "IDENTIFIER",
+        Bracketed(Ref("ExpressionSegment")),
+    )
+
+
+class ObjectReferenceSegment(ansi.ObjectReferenceSegment):
+    """A reference to an object."""
+
+    # Allow whitespace
+    match_grammar: Matchable = Delimited(
+        OneOf(Ref("SingleIdentifierGrammar"), Ref("IdentifierClauseSegment")),
+        delimiter=Ref("ObjectReferenceDelimiterGrammar"),
+        terminators=[Ref("ObjectReferenceTerminatorGrammar")],
+        allow_gaps=False,
+    )
+
+
+class DatabaseReferenceSegment(ObjectReferenceSegment):
+    """A reference to a database."""
+
+    type = "database_reference"
+
+
+class TableReferenceSegment(ObjectReferenceSegment):
+    """A reference to an table, CTE, subquery or alias."""
+
+    type = "table_reference"
+
+
+class SchemaReferenceSegment(ObjectReferenceSegment):
+    """A reference to a schema."""
+
+    type = "schema_reference"
```

Comment:
I don't think these need redefining? I can't see any edits vs the classess created in the ansi dialect, so no need to redefine. Unless I've missed something?

Reply:
@alanmcruickshank 

These were needed to inherit changes to `ObjectReferenceSegment` which has been updated, without the tests fail.

Reply:
iiiiiineresting. Good catch. Not super helpful from a development point of view, but I guess that's what it is right now.

Thanks for the contribution!